homeassistant/components/hassio/auth.py

```diff
@@ -123,6 +123,10 @@ class HassIOPasswordReset(HassIOBaseAuth):
         """Check User credentials."""
         provider = self._get_provider()
 
+        # Load data if not allready done
+        if provider.data is None:
+            await provider.async_initialize()
+
         try:
             await self.hass.async_add_executor_job(
                 provider.data.change_password, username, password
```

Comment:
This is not good, you're calling directly into `provider.data`, which means that you don't initialize (hence this PR) but also means that you don't save things ! Please add a `change_password` function to the provider and use that.

Reply:
Also, we are raising `InvalidUser` if user cannot be found, but I don't see that being caught ?

Reply:
Please add tests that test the full flow of changing passwords for an existing user and a non-existing user. And verify that the password has changed afterwards.